sphinx_gallery/scrapers.py

```diff
@@ -91,8 +102,37 @@ def matplotlib_scraper(block, block_vars, gallery_conf, **kwargs):
         the images. This is often produced by :func:`figure_rst`.
     """
     matplotlib, plt = _import_matplotlib()
+    from matplotlib.animation import FuncAnimation, ImageMagickWriter
     image_path_iterator = block_vars['image_path_iterator']
     image_rsts = []
+    # Animations first
+    if gallery_conf.get('matplotlib_animations', False):
+        for ani in block_vars['example_globals'].values():
+            if isinstance(ani, FuncAnimation) and \
+                    not hasattr(ani, '_sg_scraped'):
+                ani._sg_scraped = True  # ugly monkeypatch, but should work
+                # output the thumbnail as the image, as it will just be copied
+                # if it's the file thumbnail
+                fig = ani._fig
+                image_path_iterator = block_vars['image_path_iterator']
+                img_fname = next(image_path_iterator).replace('.png', '.gif')
```

Comment:
We should really relax the assumptions about `.png` extensions in the codebase :-)

Reply:
Someday probably :) But in the meantime, knowing it ends with `png` and `.replace`ing works okay